apps/admin-portal/src/components/identity-providers/forms/general-details-form.tsx

```diff
@@ -84,6 +87,45 @@ export const GeneralDetailsForm: FunctionComponent<GeneralDetailsFormPopsInterfa
     } = props;
 
     const [isEnable, setIsEnable] = useState<boolean>(isEnabled);
+    const [isNameValid, setIsNameValid] = useState<boolean>(true);
+    const [modifiedName, setModifiedName] = useState<string>(name);
+
+    const dispatch = useDispatch();
+
+    /**
+     * Called when name field is modified.
+     */
+    useEffect(() => {
+        setIsNameValid(false);
+        validateIdpName(modifiedName);
+    }, [modifiedName]);
+
+    /**
+     * Retrieves the list of identity providers.
+     */
+    const validateIdpName = (idpName: string) => {
+        return getIdentityProviderList(null, null, "name eq " + idpName)
+            .then((response) => {
+                setIsNameValid(response?.totalResults === 0);
+            })
+            .catch((error) => {
+                if (error.response && error.response.data && error.response.data.description) {
```

Comment:
If needed, this can be done in a single line using optional chaining.

```ts
if (error?.response?.data?.description)
```

Reply:
Fixed with c4d69b38381a5d17dd63a3c8719f26a605a679b8.